src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -96,8 +97,10 @@ type downsampler struct {
 	agg  agg
 
 	sync.RWMutex
-	metricsAppenderOpts metricsAppenderOptions
-	enabled             bool
+	metricsAppenderPool *metricsAppenderPool
+	// TODO: is this nil to start?
```

Comment:
It would be if not allocated when created yeah?

Reply:
this was more a note to myself to make sure a nil value is ok to use (which it is). essentially this is nil until the namespace has been updated via the cluster watch.